src/ChatFeed/MyMessage/index.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useState } from 'react';
+
+import { Props } from './props';
+import { styles } from './styles';
+
+import { Dot } from '../../Dot';
+
+import { FileThumb } from '../FileThumb';
+import { ImageThumb } from '../ImageThumb';
+
+import { isImage, getFileName } from '../../util/file';
+import { formatTime, getDateTime } from '../../util/dateTime';
+
+import { Row, Col, setConfiguration } from 'react-grid-system';
+
+setConfiguration({ maxScreenClass: 'xl' });
+
+export const MyMessage: React.FC<Props> = ({
+  lastMessage = null,
+  message,
+  nextMessage = null,
+  chat = null,
+  isSending = false,
+}) => {
+  const [hovered, setHovered] = useState<boolean>(false);
+
+  const topRightRadius =
+    !lastMessage || lastMessage.sender_username !== message.sender_username
+      ? '1.3em'
+      : '0.3em';
+  const bottomRightRadius =
+    !nextMessage || nextMessage.sender_username !== message.sender_username
+      ? '1.3em'
+      : '0.3em';
+
+  const borderRadius = `1.3em ${topRightRadius} ${bottomRightRadius} 1.3em`;
+  const paddingBottom =
+    !nextMessage || nextMessage.sender_username !== message.sender_username
+      ? '12px'
+      : '2px';
+
+  const text: string =
+    message.text !== null
+      ? message.text.replace(/<a /g, `<a style="color: 'white';" `)
+      : '';
+
+  const renderImages = () => {
+    const attachments =
+      message && message.attachments ? message.attachments : [];
+
+    return attachments.map((attachment, index) => {
+      const fileName = getFileName(attachment.file);
+
+      if (isImage(fileName)) {
+        return (
+          <ImageThumb
+            attachment={attachment}
+            isLoading={isSending || attachment.file === null}
+          />
+        );
+      }
+
+      return <div key={`attachment${index}`} />;
+    });
+  };
+
+  const renderFiles = () => {
+    const attachments =
+      message && message.attachments ? message.attachments : [];
+
+    return attachments.map((attachment, index) => {
+      const fileName = getFileName(attachment.file);
+
+      if (!isImage(fileName)) {
+        return (
+          <FileThumb
+            attachment={attachment}
+            isLoading={isSending || attachment.file === null}
+          />
+        );
+      }
+
+      return <div key={`attachment${index}`} />;
+    });
+  };
```

Comment:
I feel like this could be combined to a `renderAttachments` function instead of looping over the list of attachments twice you could just do if/else on the isImage and do it all in one go

Reply:
Good point. I got some time after work to do this 👍

Reply:
Actually wait there's a good reason for this too: Basically we want all the images rendered, then all the Files rendered. If we do it all in one loop then the files/images render in a mixed order which we don't want UI wise